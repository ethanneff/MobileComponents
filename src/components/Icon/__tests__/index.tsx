import React from "react";
import renderer from "react-test-renderer";
import Component from "..";
import Theme from "../../../commons/Theme";

describe("Icon Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Component name={"account"} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("renders correctly without defaults", () => {
    const component = renderer
      .create(
        <Component name={"account"} size={1} color={Theme.color.success} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it("renders correctly with badge", () => {
    const badge = 20;
    const component = renderer
      .create(
        <Component name={"account"} badge={badge} color={Theme.color.success} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("renders correctly with large badge", () => {
    const badge = 200;
    const component = renderer
      .create(
        <Component name={"account"} badge={badge} color={Theme.color.success} />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("renders correctly when hidden", () => {
    const badge = 200;
    const component = renderer
      .create(
        <Component
          hidden
          name={"account"}
          badge={badge}
          color={Theme.color.success}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
