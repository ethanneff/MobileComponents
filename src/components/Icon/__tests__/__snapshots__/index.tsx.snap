// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component renders correctly 1`] = `
<View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "hsl(199, 18%, 33%)",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Icon Component renders correctly when hidden 1`] = `null`;

exports[`Icon Component renders correctly with badge 1`] = `
<View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "hsl(84, 67%, 75%)",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderRadius": 20,
        "height": 20,
        "justifyContent": "center",
        "position": "absolute",
        "right": -10,
        "top": -10,
        "width": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Regular",
              "fontSize": 17,
              "letterSpacing": 0,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "color": "hsl(255, 100%, 100%)",
              "fontSize": 12,
              "textAlign": "center",
            },
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      20
    </Text>
  </View>
</View>
`;

exports[`Icon Component renders correctly with large badge 1`] = `
<View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "hsl(84, 67%, 75%)",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderRadius": 20,
        "height": 20,
        "justifyContent": "center",
        "position": "absolute",
        "right": -10,
        "top": -10,
        "width": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Regular",
              "fontSize": 17,
              "letterSpacing": 0,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "color": "hsl(255, 100%, 100%)",
              "fontSize": 12,
              "textAlign": "center",
            },
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      !
    </Text>
  </View>
</View>
`;

exports[`Icon Component renders correctly without defaults 1`] = `
<View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "hsl(84, 67%, 75%)",
          "fontSize": 1,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
